pkg/yurthub/proxy/remote/remote.go

```diff
@@ -132,28 +132,34 @@ func (rp *RemoteProxy) modifyResponse(resp *http.Response) error {
 
 	// cache resp with storage interface
 	if resp.StatusCode >= http.StatusOK && resp.StatusCode <= http.StatusPartialContent {
-		if rp.cacheMgr.CanCacheFor(req) {
+		var canCache bool
+		if rp.cacheMgr != nil && rp.cacheMgr.CanCacheFor(req) {
+			canCache = true
+		}
+		if rp.filterChain != nil || canCache {
```

Comment:
`ContentType` for response is not need by filterChain, so we can optimize this snippet as following.